cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -76,22 +77,37 @@ pub struct LockedInstallable {
 /// using the `pkgdb lock-flake-installable` command.
 ///
 /// The trait is also implemented by the [`InstallableLockerMock`] struct which is used for testing.
-#[enum_dispatch]
```

Comment:
blocking: resolve this discussion with @ysndr or move the change to a separate PR.

I think I prefer the macro, and @ysndr prefers the macro. I don't want to commit back and forth in code unless we come to an agreement (https://github.com/flox/flox/pull/1775#discussion_r1676325723)

Reply:
Just to be clear, my read of that comment was "it looks fine to me so I don't feel the need to do that extra work", so I just did the extra work. Not intending to sneak changes in.

Reply:
The intention with the change is just to remove magic where possible and it doesn't feel like a substantial amount of work to write one method impl by hand that just delegates to other impls (this is almost certainly what the macro is doing under the hood).

Reply:
I dont care too much,
the macro doesn't add a lot of magic, mainly just generates match statements, but not needing to write boilerplate seems worth it to me.
After all we're also not implementing standard traits and use `derive(...)` instead.
Type hints seem to be fixed now that we don't use `async_trait` anymore.